service/worker/deletenamespace/activities.go

```diff
@@ -132,7 +133,12 @@ func (a *activities) GenerateDeletedNamespaceNameActivity(ctx context.Context, n
 	const initialSuffixLength = 5
 
 	for suffixLength := initialSuffixLength; suffixLength < len(nsID.String()); suffixLength++ { // Just in case. 5 chars from ID should be good enough.
-		newName := fmt.Sprintf("%s-deleted-%s", nsName, nsID.String()[:suffixLength])
+		suffix := fmt.Sprintf("-deleted-%s", nsID.String()[:suffixLength])
+		if strings.HasSuffix(string(nsName), suffix) {
```

Comment:
```suggestion
		if strings.HasSuffix(nsName.String(), suffix) {
```
both works, this is for consistency.

Reply:
Done